### ISSUE TITLE
Breaking change in v0.8.9

### ISSUE THREAD

**Issue by mschuchard** [GoogleCloudPlatform/terraformer#631]
With the following combination, I am able to execute Terraformer against all resources in an AWS region with no errors and full expected functionality:

terraformer 0.8.8
terraform 0.12.29
terraform-provider-aws 2.70.0

Upon upgrading terraformer from 0.8.8 to 0.8.9, every attempted resource read throws the following sequence of warnings/errors (example given for VPC target):

>2020/10/26 13:59:09 WARN: Fail read resource from provider, wait 300ms before retry
...
2020/10/26 13:59:10 WARN: Fail read resource from provider, wait 300ms before retry
2020/10/26 13:59:11 Fail read resource from provider, trying import command
2020/10/26 13:59:11 Fail read resource from provider, trying import command
2020/10/26 13:59:11 rpc error: code = Canceled desc = context canceled
2020/10/26 13:59:11 rpc error: code = Canceled desc = context canceled
2020/10/26 13:59:11 ERROR: Unable to refresh resource tfer--vpc-002D-redacted
2020/10/26 13:59:11 ERROR: Unable to refresh resource tfer--vpc-002D-redacted

Unfortunately, I am unsure what other helpful context there is to provide here, so please let me know what other information is helpful to identify the breaking change. Note that `data` in Terraform and AWS CLI reads all also succeed with the identical configuration terraformer is leveraging for both 0.8.8 (success) and 0.8.9 (failure).

**Comment by sergeylanzman** [comment]
```
terraform12 init

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "aws" (hashicorp/aws) 2.70.0...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
➜  terraformer git:(master) ✗ ./terraformer import aws  -r subnet,vpc --regions eu-west-1
2020/10/31 00:34:06 aws importing region eu-west-1
2020/10/31 00:34:13 aws importing... subnet
2020/10/31 00:34:15 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXX
2020/10/31 00:34:15 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXX
2020/10/31 00:34:15 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXX
2020/10/31 00:34:15 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXXXX
2020/10/31 00:34:16 aws importing... vpc
2020/10/31 00:34:16 Refreshing state... aws_vpc.tfer--vpc-002D-XXXXXXX
2020/10/31 00:34:22 aws Connecting....
2020/10/31 00:34:22 aws save subnet
2020/10/31 00:34:22 aws save tfstate for subnet
2020/10/31 00:34:22 aws save vpc
2020/10/31 00:34:22 aws save tfstate for vpc
➜  terraformer git:(master) ✗ terraform12 version
Terraform v0.12.20
+ provider.aws v2.70.0

Your version of Terraform is out of date! The latest version
is 0.13.5. You can update by downloading from https://www.terraform.io/downloads.html
➜  terraformer git:(master) ✗ ./terraformer version
Terraformer v0.8.9 #master latest
➜  terraformer git:(master) ✗ git checkout 0.8.9
➜  terraformer git:(6868a286) ✗ go build
➜  terraformer git:(6868a286) ✗ ./terraformer import aws  -r subnet,vpc --regions eu-west-1
2020/10/31 00:40:41 aws importing region eu-west-1
2020/10/31 00:40:45 aws importing... subnet
2020/10/31 00:40:47 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXX
2020/10/31 00:40:47 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXX
2020/10/31 00:40:47 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXXXXX
2020/10/31 00:40:47 Refreshing state... aws_subnet.tfer--subnet-002D-XXXXXXXXXX
2020/10/31 00:40:48 aws importing... vpc
2020/10/31 00:40:48 Refreshing state... aws_vpc.tfer--vpc-002D-XXXXXXXXX
2020/10/31 00:40:55 aws Connecting....
2020/10/31 00:40:55 aws save subnet
2020/10/31 00:40:55 aws save tfstate for subnet
2020/10/31 00:40:55 aws save vpc
2020/10/31 00:40:55 aws save tfstate for vpc
```

Can you run with `--verbose` and try find error in output?

**Comment by lukeramsden** [comment]
Getting this too. Version v0.8.9, AWS provider 3.13.0

```
2020-11-02T11:34:05.285Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 2020/11/02 11:34:05 [DEBUG] Found KMS Alias: {
2020-11-02T11:34:05.285Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5:   AliasArn: "arn:aws:kms:eu-west-2:<account ID>:alias/aws/dynamodb",
2020-11-02T11:34:05.285Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5:   AliasName: "alias/aws/dynamodb"
2020-11-02T11:34:05.285Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: }
2020-11-02T11:34:05.288Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: panic: runtime error: invalid memory address or nil pointer dereference
2020-11-02T11:34:05.288Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4dbf937]
2020-11-02T11:34:05.288Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 
2020-11-02T11:34:05.288Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: goroutine 1933 [running]:
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsKmsAliasRead(0xc001fef580, 0x5f0a640, 0xc001f03600, 0xacc2fa0, 0x6c492c0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kms_alias.go:125 +0x3a7
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000b2cfd0, 0x7688d40, 0xc001817800, 0xc001fef580, 0x5f0a640, 0xc001f03600, 0x0, 0x0, 0x0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:280 +0x88
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000b2cfd0, 0x7688d40, 0xc001817800, 0xc001452d90, 0x5f0a640, 0xc001f03600, 0x0, 0x0, 0xc00192d380, 0x0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:554 +0x1c2
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000eec340, 0x7688d40, 0xc001817800, 0xc000de7ce0, 0xc000eec340, 0xc000eec350, 0x6c492b0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/helper/plugin/grpc_provider.go:572 +0x42f
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadResource_Handler.func1(0x7688d40, 0xc001817800, 0x68e6560, 0xc000de7ce0, 0xc001817800, 0x6091660, 0xc000de7c01, 0xc001466b40)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3276 +0x86
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve.func3.1(0x7688e00, 0xc00192d110, 0x68e6560, 0xc000de7ce0, 0xc001466b20, 0xc001466b40, 0xc00062fba0, 0x1090018, 0x6737900, 0xc00192d110)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/plugin/serve.go:76 +0x87
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadResource_Handler(0x69ca620, 0xc000eec340, 0x7688e00, 0xc00192d110, 0xc000de7c80, 0xc000eedc00, 0x7688e00, 0xc00192d110, 0xc001793810, 0x61)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3278 +0x14b
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00033fdc0, 0x76ac4e0, 0xc0008a2480, 0xc001875b00, 0xc00179a510, 0xac87d50, 0x0, 0x0, 0x0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:1171 +0x50a
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc00033fdc0, 0x76ac4e0, 0xc0008a2480, 0xc001875b00, 0x0)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:1494 +0xccd
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000564f0, 0xc00033fdc0, 0x76ac4e0, 0xc0008a2480, 0xc001875b00)
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:834 +0xa1
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-11-02T11:34:05.289Z [DEBUG] plugin.terraform-provider-aws_v3.13.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.30.0/server.go:832 +0x204
2020/11/02 11:34:05 WARN: Fail read resource from provider, wait 300ms before retry
2020-11-02T11:34:05.293Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020-11-02T11:34:05.293Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/aws/3.13.0/darwin_amd64/terraform-provider-aws_v3.13.0_x5 pid=41816 error="exit status 2"
2020/11/02 11:34:05 WARN: Fail read resource from provider, wait 300ms before retry
2020/11/02 11:34:05 WARN: Fail read resource from provider, wait 300ms before retry
2020/11/02 11:34:06 WARN: Fail read resource from provider, wait 300ms before retry
2020/11/02 11:34:06 WARN: Fail read resource from provider, wait 300ms before retry
2020/11/02 11:34:06 Fail read resource from provider, trying import command
2020/11/02 11:34:06 rpc error: code = Canceled desc = context canceled
2020/11/02 11:34:06 Refreshing state... aws_kms_alias.tfer--alias-002F-aws-002F-ebs
2020/11/02 11:34:07 WARN: Fail read resource from provider, wait 300ms before retry

```

**Comment by mschuchard** [comment]
Stack trace below. Can also confirm above comment that KMS seems to be especially problematic with Terraformer recently.

> 2020/11/02 15:39:29 Refreshing state... aws_vpc.tfer--vpc-002D-13b2ec69
2020/11/02 15:39:29 Refreshing state... aws_vpc.tfer--vpc-002D-8276ace7
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: panic: interface conversion: interface {} is nil, not *aws.AWSClient
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: goroutine 69 [running]:
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsVpcRead(0xc001548460, 0x0, 0x0, 0xc001548460, 0x0)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_vpc.go:234 +0x1475
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0008dec60, 0xc00148e6e0, 0x0, 0x0, 0xc00151ac00, 0xc00148e6e0, 0x0)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:460 +0x119
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000606958, 0x6623ea0, 0xc00151a3c0, 0xc0014681e0, 0xc000606958, 0xc00151a3c0, 0xc00164fb30)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d8
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x5a8d0c0, 0xc000606958, 0x6623ea0, 0xc00151a3c0, 0xc001468180, 0x0, 0x6623ea0, 0xc00151a3c0, 0xc001314480, 0x16d)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000c8f200, 0x6648480, 0xc000c8fc80, 0xc001310100, 0xc00088a9f0, 0x9acc190, 0x0, 0x0, 0x0)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1024 +0x4f4
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc000c8f200, 0x6648480, 0xc000c8fc80, 0xc001310100, 0x0)
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1313 +0xd97
2020-11-02T15:39:29.835-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00089c7f0, 0xc000c8f200, 0x6648480, 0xc000c8fc80, 0xc001310100)
2020-11-02T15:39:29.836-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:722 +0xbb
2020-11-02T15:39:29.836-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-11-02T15:39:29.836-0500 [DEBUG] plugin.terraform-provider-aws_v2.70.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:720 +0xa1

**Comment by mschuchard** [comment]
Is Terraformer working with AWS again in the latest release? I had not had a chance to test.

**Comment by meshuga** [comment]
Issue you raised been solved yesterday, we now import only customer keys (ref https://github.com/GoogleCloudPlatform/terraformer/blob/master/providers/aws/kms.go#L62 ), will be available in 0.8.12
